### PR TITLE
Web ant94 2021 02 01

### PR DESCRIPTION
**Pull request by naveedaz**
<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [x] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of "no breaking changes"
- [x] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [x] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [x] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  -  [x] Ensure to copy the existing version into new directory structure for first commit (including refactoring) and then push new changes including version updates in separate commits. This is required to review the changes efficiently.
  - Adding a new service

- [x] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [x] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from Breaking Change Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable or public preview version with Breaking Change Validation errors
- [ ] Updating API(s) in public preview over 1 year (refer to [Retirement of Previews](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37683/Retirement-of-Previews))

**Action**: to initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://aka.ms/Breakingchangetemplate). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by mentat9** [comment]
>         "zoneRedundant": {

Consider using a string enum, e.g. zonePolicy: { notZoned, zoneRedundant } or similar. Generally enums are more descriptive, flexible and future-proof than booleans and easier for customers to discover, understand and use.

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2021-02-01/CommonDefinitions.json:155 in 55a6949. [](commit_id = 55a6949022249a05656fbaba7643b7e727708f5c, deletion_comment = False)

**Comment by mentat9** [comment]
>             "elasticScaleEnabled": {

Similar comment here re: boolean vs. string enum. Something like scalePolicy: { PerSite, Elastic } is preferred vs. two separate booleans. Enums also allow adding members without an API version bump.

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2021-02-01/CommonDefinitions.json:221 in 55a6949. [](commit_id = 55a6949022249a05656fbaba7643b7e727708f5c, deletion_comment = False)

**Comment by mentat9** [comment]
>             "zoneRedundant": {

Consider string enum here as well.

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2021-02-01/CommonDefinitions.json:312 in 55a6949. [](commit_id = 55a6949022249a05656fbaba7643b7e727708f5c, deletion_comment = False)

**Comment by mentat9** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/networkConfig/virtualNetwork": {

Since you fleshing out the ARM resource model for this resource, consider implementing the collection GET as well to enable discovery. This would automatically give you support for Azure Policy on these resources. Note the singleton pattern for ARM resources still depends on collection GET returning an array with a single element for discovery.

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2021-02-01/WebApps.json:5945 in 55a6949. [](commit_id = 55a6949022249a05656fbaba7643b7e727708f5c, deletion_comment = False)

**Comment by mentat9** [comment]
>     "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.Web/sites/{name}/slots/{slot}/migratemysql/status": {

Similar point here: since you fleshing out the ARM resource model for this resource, consider implementing the collection GET as well to enable discovery.

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2021-02-01/WebApps.json:14868 in 55a6949. [](commit_id = 55a6949022249a05656fbaba7643b7e727708f5c, deletion_comment = False)

**Comment by mentat9** [comment]
>               "$ref": "./CommonDefinitions.json#/definitions/VnetInfo",

Should this be VnetInfoResource (the full ARM object)?

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2021-02-01/WebApps.json:23647 in 55a6949. [](commit_id = 55a6949022249a05656fbaba7643b7e727708f5c, deletion_comment = False)

**Comment by mentat9** [comment]
>             }

What will back-level clients see in place of this?

---
Refers to: specification/web/resource-manager/Microsoft.Web/stable/2021-02-01/WebApps.json:21211 in 55a6949. [](commit_id = 55a6949022249a05656fbaba7643b7e727708f5c, deletion_comment = False)

File: specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json [hunk 1]
```diff
@@ -0,0 +1,1727 @@
...
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/providers/Microsoft.CertificateRegistration/validateCertificateRegistrationInformation": {
+      "post": {
```
Comment (mentat9) [inline specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json:54]: >post [](http://example.com/codeflow?start=7&length=4)

How does this operation indicate a validation failure?

File: specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json [hunk 2]
```diff
@@ -0,0 +1,1727 @@
...
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "App Service Certificate Order is created.",
+            "schema": {
+              "$ref": "#/definitions/AppServiceCertificateOrder"
+            }
+          },
+          "201": {
```
Comment (mentat9) [inline specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json:307]: >201 [](http://example.com/codeflow?start=11&length=3)

Need to add x-ms-long-running-operation flag.

File: specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json [hunk 3]
```diff
@@ -0,0 +1,1727 @@
...
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "App Service Certificate is created.",
+            "schema": {
+              "$ref": "#/definitions/AppServiceCertificateResource"
+            }
+          },
+          "201": {
```
Comment (mentat9) [inline specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json:573]: >201 [](http://example.com/codeflow?start=11&length=3)

Need to add x-ms-long-running-operation flag.

File: specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json [hunk 4]
```diff
@@ -0,0 +1,1727 @@
...
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
```
Comment (mentat9) [inline specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json:710]: >responses [](http://example.com/codeflow?start=9&length=9)

What response indicates the email is successfully sent?

File: specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json [hunk 5]
```diff
@@ -0,0 +1,1727 @@
...
+            }
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
```
Comment (mentat9) [inline specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json:671]: >responses [](http://example.com/codeflow?start=9&length=9)

What reponse indicates the cert was successfully renewed?

File: specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json [hunk 6]
```diff
@@ -0,0 +1,1727 @@
...
```
Comment (mentat9) [inline specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json:623]: >responses [](http://example.com/codeflow?start=9&length=9)

What response indicates the cert was successfully reissued?

File: specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json [hunk 7]
```diff
@@ -0,0 +1,1727 @@
...
+          "204": {
```
Comment (mentat9) [inline specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json:624]: >204 [](http://example.com/codeflow?start=11&length=3)

Per RPC, POST should return 200 on success.

Wiki link: [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/proxy-api-reference.md#resource-action-requests](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/proxy-api-reference.md#resource-action-requests)

File: specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json [hunk 8]
```diff
@@ -0,0 +1,1727 @@
...
```
Comment (mentat9) [inline specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json:672]: >204 [](http://example.com/codeflow?start=11&length=3)

Should use 200 for success.

File: specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json [hunk 9]
```diff
@@ -0,0 +1,1727 @@
...
```
Comment (mentat9) [inline specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json:711]: >204 [](http://example.com/codeflow?start=11&length=3)

Should use 200 for success.

File: specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json [hunk 10]
```diff
@@ -0,0 +1,1727 @@
...
```
Comment (mentat9) [inline specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json:759]: >204 [](http://example.com/codeflow?start=11&length=3)

Should use 200 for success.

File: specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json [hunk 11]
```diff
@@ -0,0 +1,1727 @@
...
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CertificateRegistration/certificateOrders/{certificateOrderName}/retrieveSiteSeal": {
+      "post": {
+        "tags": [
+          "AppServiceCertificateOrders"
+        ],
+        "summary": "Verify domain ownership for this certificate order.",
```
Comment (mentat9) [inline specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json:776]: >Verify domain ownership for this certificate order. [](http://example.com/codeflow?start=20&length=51)

summary and description need to be corrected.

File: specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json [hunk 12]
```diff
@@ -0,0 +1,1727 @@
...
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.CertificateRegistration/certificateOrders/{certificateOrderName}/resendRequestEmails": {
+      "post": {
+        "tags": [
+          "AppServiceCertificateOrders"
+        ],
+        "summary": "Verify domain ownership for this certificate order.",
```
Comment (mentat9) [inline specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json:728]: >Verify domain ownership for this certificate order. [](http://example.com/codeflow?start=20&length=51)

summary and description need to be corrected.

File: specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json [hunk 13]
```diff
@@ -0,0 +1,1727 @@
...
+          "description": "Certificate Signature algorithm.",
+          "type": "string",
+          "readOnly": true
+        },
+        "issuer": {
+          "description": "Certificate Issuer.",
+          "type": "string",
+          "readOnly": true
+        },
+        "rawData": {
```
Comment (mentat9) [inline specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json:1486]: >rawData [](http://example.com/codeflow?start=9&length=7)

Is this a secret? If so, mark it with x-ms-secret.

File: specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json [hunk 14]
```diff
@@ -0,0 +1,1727 @@
...
+                "CertificateExpired",
+                "CertificateExpirationWarning",
+                "FraudDocumentationRequired",
+                "Unknown"
+              ],
+              "type": "string",
+              "readOnly": true,
+              "x-ms-enum": {
+                "name": "CertificateOrderActionType",
+                "modelAsString": false
```
Comment (mentat9) [inline specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json:1555]: >false [](http://example.com/codeflow?start=33&length=5)

Recommended value for modelAsString is true. I see this is consistently false for all enums in this PR, so presumably you have a reason for that preference.

**Comment by mentat9** [comment]
> > > ```
> > >     "zoneRedundant": {
> > > ```
> > 
> > 
> > Consider using a string enum, e.g. zonePolicy: { notZoned, zoneRedundant } or similar. Generally enums are more descriptive, flexible and future-proof than booleans and easier for customers to discover, understand and use.
> > Refers to: specification/web/resource-manager/Microsoft.Web/stable/2021-02-01/CommonDefinitions.json:155 in [55a6949](https://github.com/Azure/azure-rest-api-specs/commit/55a6949022249a05656fbaba7643b7e727708f5c). [](commit_id = [55a6949](https://github.com/Azure/azure-rest-api-specs/commit/55a6949022249a05656fbaba7643b7e727708f5c), deletion_comment = False)
> 
> The swagger changes are late for the service and the API is already live with the boolean. We can consider enums for future API versions and new properties that we introduce but is this a blocker for this change?

@naveedaz: please work with this week's on-call ARM reviewer (@chiragg4u) to complete this review.

**Comment by naveedaz** [comment]
> 23647

No. 

> > ```
> >           "$ref": "./CommonDefinitions.json#/definitions/VnetInfo",
> > ```
> 
> Should this be VnetInfoResource (the full ARM object)?
> 
> Refers to: specification/web/resource-manager/Microsoft.Web/stable/2021-02-01/WebApps.json:23647 in [55a6949](https://github.com/Azure/azure-rest-api-specs/commit/55a6949022249a05656fbaba7643b7e727708f5c). [](commit_id = [55a6949](https://github.com/Azure/azure-rest-api-specs/commit/55a6949022249a05656fbaba7643b7e727708f5c), deletion_comment = False)

No. The change to not have it be a full arm object addresses issue: https://github.com/Azure/azure-rest-api-specs/issues/14824

File: specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json (hunk 13 above)
Comment (naveedaz) [reply specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json:1486]: The issuer is not a secret.

File: specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json (hunk 14 above)
Comment (naveedaz) [reply specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json:1555]: This is not a change from previous versions. If we change modelAsString to true now is that going to cause more issues for customers with backwards compatibility?

File: specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json (hunk 1 above)
Comment (naveedaz) [reply specification/web/resource-manager/Microsoft.CertificateRegistration/stable/2021-02-01/AppServiceCertificateOrders.json:54]: Unfortunately, the current implementation only throws an exception.

**Pull request closed without merging**